### PR TITLE
fix(mdSelect): Do not close on scroll

### PR DESCRIPTION
**Pull request by FrankMeyer**
Do not close the md-select when scrolled and mouseup event is above the md-options.
Remember the mouse x-position of the mousedown event.
Take the remembered x-position instead of the mouseup-x-position to check if the user has scrolled or not.
Do not close the select menu when scrolling is detected.

See issue #4707 and #4078

### CODE REVIEW

File: src/components/select/select.js [hunk 1]
```diff
@@ -1136,11 +1142,11 @@ function SelectProvider($$interimElementProvider) {
            */
           function mouseOnScrollbar() {
             var clickOnScrollbar = false;
-            if (ev && (ev.currentTarget.children.length > 0)) {
+            if (mouseDownXPosition != null && ev && (ev.currentTarget.children.length > 0)) {
```
Comment (ThomasBurleson) [inline src/components/select/select.js]: Should we not also reset `mouseDownXPosition = null` on 'mouseup' ?

Comment (FrankMeyer) [inline src/components/select/select.js]: I htought it would not be neccessary because there is always the mousedown before the mouseup, but maybe that is not the case when the mousedown was outside the mdSelect. I'll add that.

Comment (ThomasBurleson) [inline src/components/select/select.js]: Agreed that such scenarios are edge cases. I like, however, to always *balance* my event logic: if one (1) action sets a state then a *paired* action should clear that state.

Comment (FrankMeyer) [inline src/components/select/select.js]: Completely understandable. Another question btw: Would it be a better way, or more angular way, to make a container element with a content child to have a clear distinction between mouseevents? Container mouseevent would then always originate from the scrollbar and content mouseevent always from a mdOption.

Comment (ThomasBurleson) [inline src/components/select/select.js]: That sounds like a good idea. Can you add that to this PR?

Comment (ThomasBurleson) [inline src/components/select/select.js]: @FrankMeyer - While that feature may be more complex and require significant regression issues, my intuition suggests that it may be a better long term solution.

Comment (FrankMeyer) [inline src/components/select/select.js]: Yes you are right, so should this solution be merged now and revised later on? Or should I revise it first?

Comment (ThomasBurleson) [inline src/components/select/select.js]: Let me merge this one. Then, if you finish the **container** enhancement, you can submit another PR. Does that sound reasonable?

> BTW, thanks for all your great comments, issues, research, and PR(s).

Comment (FrankMeyer) [inline src/components/select/select.js]: Yes, I can do that in the beginning of next week.

Comment (ThomasBurleson) [inline src/components/select/select.js]: I will merge this now. With your next PR, plz provide a paired test for each change (and perhaps one for this fix here). Thx again.

Comment (FrankMeyer) [inline src/components/select/select.js]: Hello ThomasBurleson,
I ve not made the check with another container Element. It was much much easier.
I skipped all the mouse position calculations and added a one-liner.
The mouse event already provided the originating element, so this was perfekt to check whether an event belongs to scrolling or not. I don't know why I ve not seen that before.

**Comment by FrankMeyer** [comment]
@ThomasBurleson ,
I have not made the check with another container Element. It was much much easier.
I skipped all the mouse position calculations and added a one-liner.
The mouse event already provided the originating element, so this was perfekt to check whether an event belongs to scrolling or not. I don't know why I ve not seen that before.
The changes are in this PR.

**Comment by EladBezalel** [comment]
Please check this against HEAD, closing for now, if it's still valid please rebase and squash your commits and i will reopen

**Comment by FrankMeyer** [comment]
The problem does not occur anymore, this can stay closed.
